flare/env.py

```diff
@@ -21,12 +21,16 @@ class AtomicEnvironment:
     :type cutoffs: np.ndarray
     """
 
-    def __init__(self, structure: Structure, atom: int, cutoffs, sweep=1):
+    def __init__(self, structure: Structure, atom: int, cutoffs):
         self.structure = structure
         self.positions = structure.wrapped_positions
         self.cell = structure.cell
         self.species = structure.coded_species
-        self.sweep_array = np.arange(-sweep, sweep+1, 1)
+
+        # Set the sweep array based on the 2-body cutoff.
+        sweep_val = ceil(cutoffs[0] / structure.max_cutoff)
+        self.sweep_val = sweep_val
+        self.sweep_array = np.arange(-sweep_val, sweep_val + 1, 1)
```

Comment:
Instead of indexing by the 2-body, (which you use because it's assumed to be the largest right?), could also just use the `max` of cutoffs. For instance, we have a 'three body' only kernel, where an unassuming user could (reasonably) think that the 2 body cutoff is a dummy value and set it to a number smaller than it should be, smaller than the 3 body cutoffs, which would be a very subtle, diabolical bug. To my knowledge we have no assert statements which guarantee the 2-body is the biggest.

Reply:
Excellent catch, I've switched to max.